solidity-v1/dashboard/src/services/threshold-authorization.service.js

```diff
@@ -0,0 +1,101 @@
+import {
+  ContractsLoaded,
+  getContractDeploymentBlockNumber,
+  getThresholdTokenStakingAddress,
+} from "../contracts"
+import { getAllOperatorStakedEventsByAuthorizer } from "./token-staking.service"
+import {
+  AUTH_CONTRACTS_LABEL,
+  TOKEN_GRANT_CONTRACT_NAME,
+} from "../constants/constants"
+import { Keep } from "../contracts"
+import { isSameEthAddress } from "../utils/general.utils"
+import { gt } from "../utils/arithmetics.utils"
+
+const fetchThresholdAuthorizationData = async (address) => {
+  if (!address) {
+    return []
+  }
+  const thresholdTokenStakingContractAddress = getThresholdTokenStakingAddress()
+  const { stakingContract, grantContract } = await ContractsLoaded
+  const keepOperatorStakedEvents = await getAllOperatorStakedEventsByAuthorizer(
+    address
+  )
+  const authorizerOperators = keepOperatorStakedEvents.map(
+    (_) => _.returnValues.operator
+  )
+  const authorizationData = []
+
+  const keepToTStakedEvents =
+    await Keep.keepToTStaking.getStakedEventsByOperator(authorizerOperators)
+
+  const operatorsStakedToT = keepToTStakedEvents.map(
+    (event) => event.returnValues.stakingProvider
+  )
+
+  const tokenGrantStakingEvents = (
+    await grantContract.getPastEvents("TokenGrantStaked", {
+      fromBlock: await getContractDeploymentBlockNumber(
+        TOKEN_GRANT_CONTRACT_NAME
+      ),
+      filter: { operator: authorizerOperators },
+    })
+  ).map((event) => {
+    return {
+      operator: event.returnValues.operator,
+    }
+  })
+
+  // Fetch all authorizer operators
+  for (let i = 0; i < authorizerOperators.length; i++) {
+    const operatorAddress = authorizerOperators[i]
+
+    const stakeParticipant = keepOperatorStakedEvents.find((event) => {
+      return isSameEthAddress(operatorAddress, event.returnValues.operator)
+    })
```

Comment:
We can iterate through the `keepOperatorStakedEvents` in the main `for` loop and get rid of this part.